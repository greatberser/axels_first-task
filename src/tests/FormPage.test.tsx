import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('FormPage', () => {
  it('should match a snapshot', () => {
    const { asFragment } = render(<div>FormPage Component</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain an expense input', () => {
    render(
      <form>
        <label htmlFor="expense">Expense</label>
        <input id="expense" type="text" />
        <button type="submit">Submit</button>
      </form>
    );
    expect(screen.getByLabelText(/expense/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument();
  });

  it('should contain a submit button', () => {
    render(
      <form>
        <label htmlFor="expense">Expense</label>
        <input id="expense" type="text" />
        <button type="submit">Submit</button>
      </form>
    );
    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument();
  });
});
