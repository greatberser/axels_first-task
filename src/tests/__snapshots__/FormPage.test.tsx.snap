// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FormPage renders without crashing 1`] = `
<DocumentFragment>
  <div>
    FormPage Component
  </div>
</DocumentFragment>
`;
